src/Event/Event.php

```diff
@@ -89,6 +94,29 @@ public function __get($attribute)
         if ($attribute === 'name' || $attribute === 'subject') {
             return $this->{$attribute}();
         }
+        if ($attribute === 'data') {
+            return $this->_data;
+        }
+        if ($attribute === 'result') {
+            return $this->_result;
+        }
+    }
+
+    /**
+     * Provides backward compatibility for write access to data and result properties.
+     *
+     * @param string $attribute Attribute name.
+     * @param mixed $value The value to set.
+     * @return void
```

Comment:
We're avoiding using `trigger_error` on deprecated things for now as it makes people really grumpy when they upgrade.

I suspect before 4.0 we'll do a release that adds `trigger_error` to all the deprecated methods.